src/transformers/pipelines/base.py

```diff
@@ -908,16 +910,37 @@ def __init__(
                 # then we should keep working
                 self.image_processor = self.feature_extractor
 
-    def save_pretrained(self, save_directory: str, safe_serialization: bool = True):
+    def save_pretrained(
+        self,
+        save_directory: Union[str, os.PathLike],
+        safe_serialization: bool = True,
+        token: Optional[Union[str, bool]] = None,
```

Comment:
Having this isn't consistent with the logic in the method 

```suggestion
```

Reply:
most `save_pretrained` methods pass the token in kwargs, even in here if I pass the token to the parameters I will need to set it up to kwargs again since the `push_to_hub` has the `token` as a parameter I think it will suffice to make it work.

fundamentally speaking it will be misleading to have the token in the `save_pretrained` method.

do let me know if you agree with this or if i should proceed with your changes

Reply:
> do let me know if you agree with this or if i should proceed with your changes

I'm a bit confused. My proposed changes are to remove this as you say:

> fundamentally speaking it will be misleading to have the token in the save_pretrained method parameters.
> most save_pretrained methods pass the token in kwargs
> even in here if I pass the token to the parameters I will need to set it up to kwargs again

so we shouldn't have this as an argument in save_pretrained. What are the changes you're proposing?

Reply:
oh, i have it as a parameter, i got confused a little, let me merge that

Reply:
this is one of the very first times contributed to a project, so apologize if I got confused a lot, I merged your changes, also thanks for the feedback